lib/block_view_message.go

```diff
@@ -825,95 +949,189 @@ func (bav *UtxoView) _connectMessagingGroup(
-	// would be tricky to impose in consensus, considering that removed users can't *forget* the messaging private key.
-	// Removing users can be facilitated in the application-layer, where we can issue a new group key and share it with
-	// all valid members.
-
-	// We will keep track of all group messaging members.
-	var messagingMembers []*MessagingGroupMember
-	// Map all members so that it's easier to check for overlapping members.
-	existingMembers := make(map[PublicKey]bool)
-
-	// Sanity-check a group's members can't contain the messagingPublicKey.
-	existingMembers[*messagingPublicKey] = true
-
-	// If we're adding more group members, then we need to make sure there are no overlapping members between the
-	// transaction's entry, and the existing entry.
-	if existingEntry != nil && !existingEntry.isDeleted {
-		// We make sure we'll add at least one messaging member in the transaction.
-		if len(txMeta.MessagingGroupMembers) == 0 {
-			return 0, 0, nil, errors.Wrapf(RuleErrorMessagingKeyDoesntAddMembers,
-				"_connectMessagingGroup: Can't update a messaging key without any new recipients")
+	// Check what type of operation we are performing.
+	switch messagingGroupOperation {
+	case MessagingGroupOperationAddMembers:
+		// In DeSo V3 Messages, a messaging key can initialize a group chat with more than two parties. In group chats, all
+		// messages are encrypted to the group messaging public key. The group members are provided with an encrypted
+		// private key of the group's messagingPublicKey so that each of them can read the messages. We refer to
+		// these group members as messaging members, and for each member we will store a MessagingMember object with the
+		// respective encrypted key. The encrypted key must be addressed to a registered groupKeyName for each member, e.g.
+		// the base or the default key names. In particular, this design choice allows derived keys to read group messages.
+		//
+		// A MessagingGroup transaction can either initialize a groupMessagingKey or add more members. In the former case,
+		// there will be no existing MessagingGroupEntry; however, in the latter case there will be an entry present in DB
+		// or UtxoView. When adding members, we need to make sure that the transaction isn't trying to change data about
+		// existing members. An important limitation is that the current design doesn't support removing recipients. This
+		// would be tricky to impose in consensus, considering that removed users can't *forget* the messaging private key.
+		// Removing users can be facilitated in the application-layer, where we can issue a new group key and share it with
+		// all valid members.
+
+		// Map all members so that it's easier to check for overlapping members.
+		existingMembers := make(map[PublicKey]bool)
+
+		// Sanity-check a group's members can't contain the messagingPublicKey.
+		existingMembers[*messagingPublicKey] = true
+
```

Comment:
if you want to be even more memory efficient, you can use an empty struct. It doesn't look like we're checking for the "true" value in this map, so existence itself implies a "truthy" value. Empty structs cost 0 bytes and effectively make a map a hash set
```suggestion
		// Map all members so that it's easier to check for overlapping members.
		existingMembers := make(map[PublicKey]struct{})

		// Sanity-check a group's members can't contain the messagingPublicKey.
		existingMembers[*messagingPublicKey] = struct{}{}

```